packages/subiquity_client/lib/subiquity_client.dart

```diff
@@ -23,9 +24,13 @@ String _formatResponseLog(String method, String response) {
 
 class SubiquityClient {
   late HttpUnixClient _client;
+  final _completer = Completer<bool>();
+
+  Future<bool> get isOpen => _completer.future;
 
   void open(String socketPath) {
     log.info('Opening socket $socketPath');
+    _completer.complete(true);
```

Comment:
Does it make any difference if this is done as the first thing or the last thing in this function? Wondering if there's a case where something could react to this before `_client` is set (and any potential future code that is later added).

Reply:
true, completing it after opening the client is the correct order, even if it didn't seem to make a difference to the UI, probably because it's asynchronous